engine/client/client.go

```diff
@@ -261,7 +272,11 @@ func Connect(ctx context.Context, params Params) (_ *Client, _ context.Context,
 	bkSession.Allow(authprovider.NewDockerAuthProvider(config.LoadDefaultConfigFile(os.Stderr), nil))
 
 	// host=>container networking
-	c.Recorder = progrock.NewRecorder(progMultiW)
+	if c.ProgrockParent != "" {
```

Comment:
It seems like this is what we were doing before so not a big deal, but I did just realize we set `c.Recorder` both here and above on line 147 and am confused now why (or if we actually need to in both places).

Reply:
Yeah I can't see a good reason to, and I obviously just kept pattern-matching here, guessing it ended up this way through little changes over time. I think at one point there was a local `recorder :=` var involved, and it was consolidated to just `c.Recorder` since it ended up being assigned to it anyway. Will clean up